include/deal.II/base/std_cxx14/memory.h

```diff
@@ -0,0 +1,45 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+#ifndef dealii__cxx14_memory_h
+#define dealii__cxx14_memory_h
+
+#include <deal.II/base/config.h>
+
+#include <memory>
+
+DEAL_II_NAMESPACE_OPEN
+namespace std_cxx14
+{
+#ifdef DEAL_II_WITH_CXX14
+
+  using std::make_unique;
+#else
+
+#  include <type_traits>
```

Comment:
Bad idea -- don't put a `#include` into an open namespace. You should `#include` it at the top of the file (surrounded by the appropriate `#ifdef`).